### PR TITLE
Pin hdf5 to 1.10.3

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -341,7 +341,7 @@ harfbuzz:
 hdf4:
   - 4.2
 hdf5:
-  - 1.10.2
+  - 1.10.3
```
Comment (ocefpaf) [inline recipe/conda_build_config.yaml]: We don't pin software here unless it **must** diverge from the global pinning.

You should send a PR to https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L344 instead so all packages will get re-pinned.

Comment (ocefpaf) [reply recipe/conda_build_config.yaml]: Jeez I clicked on your hdf5 PR and commented here. Sorry. Disregard what I wrote above. This is correct!

**Comment by ocefpaf** [comment]
Please bump the build number to get a new package.

**Comment by scopatz** [comment]
Thanks @tschoonj and @ocefpaf!

**Comment by tschoonj** [comment]
Cool: so do I open a PR in h5py-feedstock to ensure a new build gets made against hdf5 1.10.3 or does this happen automagically?

**Comment by scopatz** [comment]
Yeah, you can open an rerender PR if you want to ensure this version of hdf5 is used.  Otherwise it will happen automatically whenever h5py gets rebuilt next.

**Pull request merged**